blog/_posts/2023-12-20-gsoc-report.md

```diff
@@ -0,0 +1,46 @@
+In this project, Dang Trung Hieu worked under the mentorship of Arman Bilge, Christopher Davenport and Diego Alonso to implement a [WebSocket client](https://github.com/http4s/http4s/pull/7196) for the [http4s](https://github.com/http4s/http4s) Ember backend. The goal was to extend the capabilities of the http4s library, a Scala HTTP library, with robust WebSocket support to enable real-time communication features in web applications.
+
+The project's success lies in the development of a fully functional WebSocket client that integrates seamlessly with the existing http4s Ember framework. This integration enhances the library's utility in building modern, real-time web applications, thereby enriching the Scala ecosystem.
+
+## Data Visualization with Doodle
+Danylo Ostapchenko worked on this project under the mentorship of Noel Welsh to enhance the [Doodle library](https://www.creativescala.org/doodle/), a tool for compositional vector graphics in Scala. The aim was to improve the library's data visualization capabilities, making it more versatile and user-friendly for developers.
+
+Danylo's work led to significant enhancements in the Doodle library, including new features and optimizations that make data visualization more intuitive and powerful. These improvements are expected to benefit developers seeking to integrate complex graphical representations into their Scala applications. You can read more about Danylo's work in his [final report](https://danielost.github.io/posts/gsoc23/)
+
+## Async-Shifted High-Level APIs
+Olena Kravchenko worked under the mentorship of Ruslan Shevchenko on a project focused on creating an 'async-shifted' version of high-order functions in Scala. The project aimed to streamline asynchronous programming in Scala by making high-level APIs more efficient and easier to use in asynchronous contexts. During this project, Olena pushed the boundaries of the dotty-cps-async library, which provides a framework for asynchronous programming in Scala based on the CPS process algebra. You can read more about her work in the [final report](https://docs.google.com/document/d/1V5harGKumEVvD1975IV-YVEKzyTW1heACfMCg-TdROM/edit).
+
+## Enhancing Scala 3's Macro System
+Yuito Murase worked under the mentorship of Nicolas Stucki to enhance the macro system in Scala 3, specifically in the area of pattern matching. The project's goal was to improve quote pattern matching in Scala 3 macros to add better support for matching against types.
+
+Yuito's efforts led to significant improvements in the Scala 3 macro system, particularly in making pattern matching more versatile and robust. These enhancements are crucial for developers who utilize metaprogramming in Scala, allowing for more sophisticated and efficient code generation. You can read more about the work that was done in the [final report](https://gist.github.com/zeptometer/ad414dda221a9f9f99e4135b6b5c1c3c).
+
+## Breaking the Thread Barrier: Boosting I/O Performance in Scala
```

Comment:
@antoniojimeneznieto, @armanbilge, could you please review this section? Feel free to suggest changes, or rewrite it to your liking altogether!

Reply:
Daniel Spiewak is missing from the list of mentors. Besides that, it looks good to me. 🙂